velox/exec/SortBuffer.cpp

```diff
@@ -106,6 +107,13 @@ void SortBuffer::noMoreInput() {
   velox::common::testutil::TestValue::adjust(
       "facebook::velox::exec::SortBuffer::noMoreInput", this);
   VELOX_CHECK(!noMoreInput_);
+
+  // It may trigger spill, make sure it's triggered before noMoreInput_ is set.
+  if (spiller_ == nullptr) {
+    memory::ReclaimableSectionGuard guard(nonReclaimableSection_);
+    pool_->maybeReserve(1 * 1024 * 1024 * 1024L);
```

Comment:
What's the consideration of reserving exact 1GiB here? Can we do a estimation of how much of memory will be used?

Reply:
We shall reserve based on the actual memory estimation here. Also put into a utility say ensureSortFits? Thanks!